BUGS-6304 Update transaction counts after validation trim

Cardds opened the pull request:
When the validations run with the 'truncate' option on, the transaction
ST values are not updated with the correct number of segments. By
calling `prepare!` when the truncation option has been turned on, it
updates the particular segment (transaction, envelope, or otherwise)
with the new number.

Dev note: it appears that some of the test case data that is listed as
being valid is not valid, e.g. some ST01 values are either nonexistent
when they are required for the particular format, or they have
inaccurate values.

Commit:
BUGS-6304 Update transaction counts after validation trim

When the validations run with the 'truncate' option on, the transaction
ST values are not updated with the correct number of segments. By
calling `prepare!` when the truncation option has been turned on, it
updates the particular segment (transaction, envelope, or otherwise)
with the new number.

Dev note: it appears that some of the test case data that is listed as
being valid is not valid, e.g. some ST01 values are either nonexistent
when they are required for the particular format, or they have
inaccurate values.

scottkf reviewed (changes requested):
@Cardds can you please add a corresponding spec for this?

Commit:
BUGS-6453 Default truncation to `true`

Defaulting truncation to `false` allows the EDI validation to
essentially ignore EDI spec. In general, the EDI specification should
not be ignored.

Dev Note: Ideally, the trunctation should be separated from the
validation, as validation should be a read-only operation, and the
trunctation should happen regardless.

Commit:
BUGS-6304 Ensure builder trims to match grammar

Adds a spec test to ensure that the builder trims the transaction amount
after a validation that trims the number of records

Cardds commented:
@scottkf Three additional specs to check this parameter and associated functionality.

Note that this is also addresses BUGS-6453 with a hotfix in a separate commit; more work may need to be done on BUGS-6453, but this should lower the priority.

scottkf reviewed (changes requested):
@Cardds bump the version and adjust in the broker please

Commit:
BUGS-6304 Increment patch version

Cardds commented:
@scottkf Patch # incremented in version.

scottkf reviewed (approved):
@Cardds good to merge
